WalletWasabi/WabiSabi/Models/DecompositionAlgs/RestrictedSetDecomposer.cs

```diff
@@ -0,0 +1,56 @@
+using NBitcoin;
+using System.Linq;
+
+namespace WalletWasabi.WabiSabi.Models.DecompositionAlgs
+{
+	public class RestrictedSetDecomposer
+	{
+		private Decomposition Decomposition { get; } = new();
+
+		public RestrictedSetDecomposer()
+		{
+			Decomposition.Extend(new Coin() { Amount = Money.Zero });
+		}
+
+		public virtual Money? PickNextAmount(Money remaining, Money dustThreshold)
+		{
+			return LargestDenomBelowIncl(remaining)?.Amount;
+		}
+
+		public void Decompose(Money amount, Money dustThreshold, FeeRate feeRate)
```

Comment:
`feeRate` also looks like a candidate for a property or field.

Reply:
yes